presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionarySelectiveStreamReader.java

```diff
@@ -278,13 +278,14 @@ private void openRowGroup()
         if (!dictionaryOpen && dictionarySize > 0) {
             if (dictionary.length < dictionarySize) {
                 dictionary = new long[dictionarySize];
+                systemMemoryContext.setBytes(sizeOf(dictionary));
             }
 
             LongInputStream dictionaryStream = dictionaryDataStreamSource.openStream();
             if (dictionaryStream == null) {
                 throw new OrcCorruptionException(streamDescriptor.getOrcDataSourceId(), "Dictionary is not empty but data stream is not present");
             }
-            dictionaryStream.nextLongVector(dictionarySize, dictionary);
+            dictionaryStream.next(dictionary, dictionarySize);
```

Comment:
What's the reason for this change? Perhaps, move it into a separate commit and explain it in commit messase.